src/data/mystery-encounters/sleeping-snorlax.ts

```diff
@@ -0,0 +1,127 @@
+import BattleScene from "../../battle-scene";
+import {
+  EnemyPartyConfig,
+  initBattleWithEnemyConfig,
+  leaveEncounterWithoutBattle,
+  pushDialogueTokensFromPokemon,
+  setEncounterRewards,
+  showEncounterText
+} from "#app/utils/mystery-encounter-utils";
+import MysteryEncounter, {MysteryEncounterBuilder} from "../mystery-encounter";
+import * as Utils from "../../utils";
+import {MysteryEncounterType} from "../enums/mystery-encounter-type";
+import {MoveRequirement, StatusEffectRequirement, WaveCountRequirement} from "../mystery-encounter-requirements";
+import {MysteryEncounterOptionBuilder} from "../mystery-encounter-option";
+import {
+  modifierTypes
+} from "#app/modifier/modifier-type";
+import PokemonSpecies, { PokemonForm, SpeciesFormKey } from "../pokemon-species";
+import { Type } from "../type";
+import { Abilities } from "../enums/abilities";
+import { Species } from "../enums/species";
+import { GrowthRate } from "../exp";
+import { StatusEffect } from "../status-effect";
+import { Moves } from "../enums/moves";
+import { SummonPhase } from "#app/phases";
+
+export const SleepingSnorlaxEncounter: MysteryEncounter = new MysteryEncounterBuilder()
+  .withEncounterType(MysteryEncounterType.SLEEPING_SNORLAX)
+  .withIntroSpriteConfigs([]) // Set in onInit()
+  .withSceneRequirement(new WaveCountRequirement([10, 180])) // waves 2 to 180
+  .withCatchAllowed(true)
+  .withOnInit((scene: BattleScene) => {
+    const instance = scene.currentBattle.mysteryEncounter;
+    pushDialogueTokensFromPokemon(instance);
+    console.log(instance);
+    const availablePartyMembers = scene.getParty().filter(p => p.isAllowedInBattle());
+    if (!availablePartyMembers[0].isOnField()) {
+
+      scene.pushPhase(new SummonPhase(scene, 0));
+    }
+
+    // Calculate boss mon
+    const bossSpecies = new PokemonSpecies(Species.SNORLAX, 1, false, false, false, "Sleeping Pokémon", Type.NORMAL, null, 2.1, 460, Abilities.IMMUNITY, Abilities.THICK_FAT, Abilities.GLUTTONY, 540, 160, 110, 65, 65, 110, 30, 25, 50, 189, GrowthRate.SLOW, 87.5, false, true,
+      new PokemonForm("Normal", "", Type.NORMAL, null, 2.1, 460, Abilities.IMMUNITY, Abilities.THICK_FAT, Abilities.GLUTTONY, 540, 160, 110, 65, 65, 110, 30, 25, 50, 189, false, null, true),
+      new PokemonForm("G-Max", SpeciesFormKey.GIGANTAMAX, Type.NORMAL, null, 35, 460, Abilities.IMMUNITY, Abilities.THICK_FAT, Abilities.GLUTTONY, 640, 200, 130, 85, 75, 130, 20, 25, 50, 189));
+    const config: EnemyPartyConfig = {
+      levelAdditiveMultiplier: 4,
+      pokemonBosses: [bossSpecies],
+      status: StatusEffect.SLEEP
+    };
+    instance.enemyPartyConfigs = [config];
+
+    // Calculate item
+    // 1-60 ULTRA, 60-120 ROGUE, 120+ MASTER
+
+    const item = modifierTypes.LEFTOVERS;
```

Comment:
I think it's very flavorful to get leftovers in the Snorlax Event, but leftovers, even if it's a rogue-tier item, is super super good. For reference, we give a choice of a random rogueball item (which might be very well 5x rogue-balls, more often than not) in the Mysterious Challenger hardest fight, which is near run-ending in many cases.

So if we're adding a Snorlax fight, it better be very hard. If it's asleep, the Snorlax could be omniboosted x2. Or, another thing I can think of is to have it be 70% great-ball item tier, 30% leftovers. Less flavorful, but more balanced.

levelAdditivemultiplier being at 4 means that by wave 150 you might be fighting something above level 200.

Reply:
good notes on balance here, I think that's fair